### PR TITLE
developer-stats - Implements developer stats calculations

### LINKED ISSUE #11: Calculate developers stats (about section)
With all the data from [repos](developersdo.github.io/opensource/data/repos.json) and [user](developersdo.github.io/opensource/data/users.json)

1. Get a total of developers and how many of them have more than 10 repositories.

2. Get a total of repositories and the average of repositories of each user. Also get all repositories with more than 1 star.

3. We need to extract all the programming languages used across all repositories.

image

### PR DESCRIPTION
**Pull request by vyzaldysanchez**
Implements calculations for developers stats in the About page.

Closes #11

### CODE REVIEW

**Comment by vyzaldysanchez** [comment]
To move forward and if we don't want to re-do requests, we might want to start fetching developers/repositories data on page load instead.

**Comment by DannyFeliz** [comment]
> To move forward and if we don't want to re-do requests, we might want to start fetching developers/repositories data on page load instead.

I was writing a comment to mention this.

Yep, we need to fetch it on page load and show the full-screen loading, could do that?

**Comment by vyzaldysanchez** [comment]
Added the fetch on page load @DannyFeliz

File: src/App.vue [hunk 1]
```diff
@@ -28,11 +28,14 @@ export default {
...
-    //   this.$store.dispatch("App/setLoading", false);
-    // }, 5000);
+    this.$store.dispatch("App/setLoading", true);
+
+    this.$store.dispatch("About/getDevelopers");
+    this.$store.dispatch("About/getRepositories");
+
+    setTimeout(() => {
+      this.$store.dispatch("App/setLoading", false);
+    }, 5000);
```
Comment (DannyFeliz) [inline src/App.vue:38]: I think we should remove the loading screen once the actions finish, not after a fixed period of time. We can use a `Promise.all` and start returning a promise in both actions.
```suggestion
    await Promise.all([
      this.$store.dispatch("About/getDevelopers"),
      this.$store.dispatch("About/getRepositories")
    ])
    this.$store.dispatch("App/setLoading", false);
```

File: src/store/About.js [hunk 2]
```diff
@@ -14,24 +16,64 @@ const state = {
...
+  },
+
+  async getDevelopers({ commit }) {
+    const developers = await API.getDevelopers().catch(e => {
+      console.error(e);
+
+      return [];
+    });
+
+    commit("SET_DEVELOPERS", developers);
```
Comment (DannyFeliz) [inline src/store/About.js:52]: To be able to wait for this to finish on page load, we need to return a promise.
```suggestion
	return new Promise((resolve) => {
	    commit("SET_DEVELOPERS", developers);
	    resolve();
	})
```

File: src/store/About.js [hunk 3]
```diff
@@ -14,24 +16,64 @@ const state = {
...
+  },
+
+  async getRepositories({ commit }) {
+    const repositories = await API.getRepositories().catch(e => {
+      console.error(e);
+
+      return [];
+    });
+
+    commit("SET_REPOSITORIES", repositories);
```
Comment (DannyFeliz) [inline src/store/About.js:62]: To be able to wait for this to finish on page load, we need to return a promise.
```suggestion
	return new Promise((resolve) => {
	    commit("SET_REPOSITORIES", repositories);
	    resolve();
	})
```

File: src/util/api.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+const API_ENDPOINT = "https://developersdo.github.io/opensource/data";
+
+export default {
+  getDevelopers() {
+    return axios.get(`${API_ENDPOINT}/users.json`).then(({ data }) => data);
+  },
+
+  getRepositories() {
+    return axios.get(`${API_ENDPOINT}/repos.json`).then(({ data }) => data);
+  }
```
Comment (DannyFeliz) [inline src/util/api.js:12]: You can define a base URL like this.
```js
axios.defaults.baseURL = "https://developersdo.github.io/opensource/data";
```
Now you can do this:
```js
axios.get("/users.json").then(() => ...)
```

File: src/store/About.js (hunk 2 above)
Comment (vyzaldysanchez) [reply src/store/About.js:52]: This function is already async, there's no need to wrap it up in a Promise object.

File: src/store/About.js (hunk 3 above)
Comment (vyzaldysanchez) [reply src/store/About.js:62]: This function is already async, there's no need to wrap it up in a Promise object.

File: src/store/About.js (hunk 2 above)
Comment (DannyFeliz) [reply src/store/About.js:52]: I didn't notice :ok_hand:

File: src/store/About.js (hunk 3 above)
Comment (DannyFeliz) [reply src/store/About.js:62]: I didn't notice, please ignore ok_hand

File: src/util/api.js (hunk 4 above)
Comment (vyzaldysanchez) [reply src/util/api.js]: Done

File: src/App.vue (hunk 1 above)
Comment (vyzaldysanchez) [reply src/App.vue]: Done, though the linter didn't allowed to be to style the promise all that way...

**Comment by DannyFeliz** [comment]
This PR enables #10

**Pull request merged**